source/Server/AdoClients/AdoApiClient.cs

```diff
@@ -201,5 +203,35 @@ public SuccessOrErrorResult<WorkItemLink[]> GetBuildWorkItemLinks(AdoBuildUrls a
                 .ToArray();
             return SuccessOrErrorResult.Conditional(validWorkItemLinks, workItemLinks);
         }
+
+        public SuccessOrErrorResult<string[]> GetProjectList(AdoUrl adoUrl, string personalAccessToken = null, bool testing = false)
+        {
+            var (status, jObject) = client.Get($"{adoUrl.OrganizationUrl}/_apis/projects?api-version=4.1",
+                personalAccessToken ?? GetPersonalAccessToken(adoUrl));
+
+            if (!status.IsSuccessStatusCode())
+            {
+                return SuccessOrErrorResult.Failure($"Error while fetching project list from Azure DevOps: {status.ToDescription(jObject, testing)}");
+            }
+
+            return jObject["value"]
+                .Select(p => p["name"].ToString())
+                .ToArray();
+        }
+
+        public SuccessOrErrorResult<int[]> GetBuildList(AdoProjectUrls adoProjectUrls, string personalAccessToken = null, bool testing = false)
+        {
+            var (status, jObject) = client.Get($"{adoProjectUrls.ProjectUrl}/_apis/build/builds?api-version=4.1",
```

Comment:
I'm assuming we have to hard code API versions, e.g. we can't **not** specify it (or use something like `latest`) and it still works?

I ask because we _currently_ specify 2 different API versions, which I'm assuming is because the `workitems` API is newer (or just further along version-wise) than the `projects`/`builds` APIs...